### PR TITLE
/api/articles done with date in descending order

### PR DESCRIPTION
**Pull request by SirNesy**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -43,3 +43,34 @@ describe(" GET /api/topics ", () => {
...
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            comment_count: expect.any(Number),
+          });
+        });
+      });
+  });
```
Comment (sirKevlar) [inline __tests__/app.test.js:68]: We need an extra test here to check if the articles are ordered correctly (date descending)

Comment (sirKevlar) [reply __tests__/app.test.js:68]: Hint: jest sorted will be useful here

Comment (SirNesy) [reply __tests__/app.test.js:68]: okay KEV. ON IT

File: app.js [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
 
 app.get("/api/topics", getTopics);
 
+app.get("/api/articles", getArticles);
+
 app.all("/*", (req, res) => {
   res.status(404).send({ msg: "invalid URL!" });
 });
+app.use((err, req, res, next) => {
+  if (err.status && err.status) {
```
Comment (Geo-Beard) [inline app.js]: Not sure why you have err.status && err.status?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -43,3 +43,51 @@ describe(" GET /api/topics ", () => {
...
+  test("GET - status 200, responds with a sorted order value ", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at")
```
Comment (Geo-Beard) [inline __tests__/app.test.js]: Don't need to test with a query - queries come in at ticket 10. This test is fine if you ".get("/api/articles") - without the query.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -43,3 +43,51 @@ describe(" GET /api/topics ", () => {
...
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("GET - status 400, Invalid sort query ", () => {
```
Comment (Geo-Beard) [inline __tests__/app.test.js]: Don't need to test queries as you shouldn't be implementing queries until ticket 10.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -43,3 +43,51 @@ describe(" GET /api/topics ", () => {
...
+    return request(app)
+      .get("/api/articles?sort_by=created_at; DROP TABLE articles")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid sort query!");
+      });
+  });
+  test("GET - status:404, not found", () => {
```
Comment (Geo-Beard) [inline __tests__/app.test.js:78]: This is OK to test but isn't 100% necessary as you will catch any invalid paths with your previous ticket tests.

File: model/news.model.js [hunk 6]
```diff
@@ -5,3 +5,21 @@ exports.selectedTopics = () => {
     return response.rows;
   });
 };
+
+exports.fetchedArticles = (sort_by = "created_at") => {
```
Comment (Geo-Beard) [inline model/news.model.js:9]: Shouldn't be implementing any sort of query functionality until ticket 10.

File: __tests__/app.test.js [hunk 7]
```diff
@@ -43,3 +43,51 @@ describe(" GET /api/topics ", () => {
...
 });
+
+describe("/api/articles", () => {
+  test("GET - status:200, responds with an article array of object", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeInstanceOf(Array);
```
Comment (Geo-Beard) [inline __tests__/app.test.js:55]: Check your array length here as you're using a forEach.

**Comment by Geo-Beard** [comment]
Quite a few points to work through - overall looks good, but you've implemented a lot of query functionality that shouldn't be implemented until ticket 10. I suggest removing this functionality for this ticket but keeping the code you've written in a scrap file somewhere (remember to add your scrap file to your git ignore). All you need is to make sure that your articles are coming back by a default DESC - you don't need to add the query option yet. The code and testing looks good - so remember it for when you get to the right ticket :)

**Pull request merged**